app/services/edit_defect.rb

```diff
@@ -32,6 +36,14 @@ def set_target_completion_date
 
     day, month, year = date_parts.map(&:to_i)
     date = Date.new(year, month, day)
-    defect.set_completion_date(date)
+    defect.set_target_completion_date(date)
+  end
+
+  def set_actual_completion_date
+    date_parts = actual_completion_date.values_at(:day, :month, :year)
+    return unless date_parts.all?(&:present?)
+
+    day, month, year = date_parts.map(&:to_i)
+    defect.actual_completion_date = Date.new(year, month, day)
```

Comment:
Direct assignment here vs method call at the end of the `set_target_completion_date` method seems inconsistent.